google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java

```diff
@@ -432,6 +445,9 @@ TransactionSelector getTransactionSelector() {
             // transactionIdFuture until an actual transactionId is available.
             if (transactionIdFuture == null) {
               transactionIdFuture = SettableApiFuture.create();
+              if (trackTransactionStarter) {
+                transactionStarter = new Exception("Requesting new transaction");
```

Comment:
This will create an `Exception` with the call stack of the statement that _should_ return a transaction id. We keep track of this to be able to include it in any exception that could be thrown by later statements that timeout while waiting for this transaction id.